src/arrayset-mutable.js

```diff
@@ -27,6 +27,12 @@ const minus = (a, b, compare) => {
   }
   return a
 }
+const fromArray = a => {
+  for(let i=a.length-1; i>=0; i--) {
+    if(a.slice(0, i).includes(a[i])) a.splice(i, 1)
+  }
+  return a
+}
```

Comment:
This algorithm is very clever...but I fear also very inefficient.  And while I don't normally prematurely optimize, this is a library so I don't know what applications it will be used in, and it _could_ be critical path algorithms.  So I feel a certain responsibility to at least consider performance.

How did I know this was inefficient?  Slice and splice are both expensive operations, relatively speaking.

This is one of those instances where my gut told me that the built-in JS libraries are probably pretty heavily optimized, so my instinct is that just using the property of the `Set` data type was the best (most efficient) bet.  But I wanted to verify, so I wrote the following perf test harness:

```js
const implementations = [
  // native JS library
  a => [...new Set(a)],
  // no array copying
  a => {
    const s = []
    for(let i=0; i < a.length; i++) {
      if(!a.includes(a[i], i + 1)) s.push(a[i])
    }
    return s
  },
  // slice/splice algorithm
  a => {
    for(let i=a.length - 1; i >= 0; i--) {
      if(a.slice(0, i).includes(a[i])) a.splice(i, 1)
    }
    return a
  },
]

const N = 100e3
const M = 100     // this will result in lots of duplicates
const arr = new Array(N).fill(null).map(() => Math.floor(Math.random()*M))

console.log(`N = ${N}\n`)

const results = implementations.forEach((f, i) => {
  const start = Date.now()
  f(arr)
  console.log(`implementation ${i}: ${Date.now() - start} ms`)
})
```

And the results were:

```
N = 100000

implementation 0: 2 ms
implementation 1: 20 ms
implementation 2: 20501 ms
```

So I think we should use implementation 0!